Automattic/jetpack
# Fix ESLint and prettier integration for VS Code

Fixes #20996

#### Changes proposed in this Pull Request:

* This PR makes `.prettierrc.js` in root directory to be a generic one
* Moves Svelte specific prettier config file to `plugins/boost`
* Fixes ESLint integration by using [@rushstack/eslint-patch](https://www.npmjs.com/package/@rushstack/eslint-patch) as a workaround.


#### Jetpack product discussion


p9dueE-4E8-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






Before following the steps in VS Code, please ensure that you
- Have [VS Code ESLint extension ](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) and [Prettier for VS Code](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) active.
- Have these settings in settings.json (`Cmd + Shift + P` > `Preferences: Open Settings (JSON)`)
```json
  "[javascript][javascriptreact][typescript][typescriptreact]": {
    "editor.formatOnSave": false,
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
  "editor.codeActionsOnSave": {
    "source.fixAll.eslint": true,
    "source.fixAll.tslint": true,
    "source.fixAll.stylelint": true
  },
```

##### Steps

* Boot up the PR and run `pnpm install`
* Reload VS Code (`Cmd + Shift + P` > `Developer: Reload Window`)
* Open any file inside `plugins/jetpack/_inc/client/` e.g. `projects/plugins/jetpack/_inc/client/at-a-glance/activity.jsx` 
* Mess up the formatting. For example select some text and run `Outdent line` command (`Cmd + [`) a few times.
* Or add too many empty lines to the file, remove some trailing bracket spaces etc.
* Open problems view (`Cmd + Shift + P` > `Problems: Focus on Problems View`)
* Confirm that you see the problems reported by ESLint.
* Save Changes (`Cmd + S`)
* Confirm that the file is formatted as expected


<img width="660" alt="Screenshot 2022-03-23 at 4 14 22 PM" src="https://user-images.githubusercontent.com/18226415/159682043-0101a188-b5b6-43bd-8684-e0c3eeaa4431.png">
